### ISSUE TITLE
Rule overlap: basic auth rule too generic, conflicts with database connection string rule

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1100]
## Problem Description

The `@secretlint/secretlint-rule-basicauth` rule is currently too generic and overlaps with the newly added `@secretlint/secretlint-rule-database-connection-string` rule. This creates redundant detection and potential conflicts.

### Basic Auth Rule
- **Pattern**: `(?<protocol>(:?[-a-zA-Z0-9_]{1,256})):\/\/(?<user>[-a-zA-Z0-9_]{1,256}):(?<password>[-a-zA-Z0-9_]{1,256})@[-a-zA-Z0-9%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b`
- **Scope**: Detects ANY protocol with `user:password@host` format
- **Intent**: Basic authentication for HTTP/HTTPS URLs

### Database Connection String Rule  
- **Patterns**: Specific patterns for `mongodb://`, `mysql://`, `postgresql://`, etc.
- **Scope**: Database-specific connection strings only
- **Features**: Advanced false positive prevention, entropy analysis, placeholder detection

## Overlap Examples

The following URLs are detected by **both rules**:
```
mongodb://user:pass@cluster.mongodb.net/database
mysql://admin:secret@db.example.com:3306/mydb
postgresql://user:pass@postgres.example.com:5432/database
```

## Proposed Solution

Restrict the basic auth rule to HTTP-related protocols only:

### Option 1: Limit to HTTP/HTTPS/FTP protocols
```javascript
const HTTP_BASIC_AUTH_PATTERN = 
  /(?<protocol>(?:https?|ftp)):\/\/(?<user>[-a-zA-Z0-9_]{1,256}):(?<password>[-a-zA-Z0-9_]{1,256})@[-a-zA-Z0-9%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b/g;
```

### Option 2: Exclude database protocols
```javascript
const NON_DB_PROTOCOL_PATTERN = 
  /(?<protocol>(?\!mongodb|mysql|postgresql|postgres)[-a-zA-Z0-9_]{1,256}):\/\/(?<user>[-a-zA-Z0-9_]{1,256}):(?<password>[-a-zA-Z0-9_]{1,256})@[-a-zA-Z0-9%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b/g;
```

### Option 3: Add configuration option
Allow users to specify which protocols should be detected by the basic auth rule.

## Benefits

1. **Clearer separation of concerns**: Basic auth rule focuses on HTTP authentication, database rule handles database credentials
2. **Reduced false positives**: No duplicate detection of the same credentials
3. **Better user experience**: Users get more specific and actionable messages
4. **Maintained backward compatibility**: HTTP/HTTPS basic auth detection remains intact

## Impact Assessment

- **Low risk**: Most real-world basic auth usage is HTTP/HTTPS/FTP
- **High benefit**: Eliminates confusion between different types of credentials
- **Easy migration**: Existing users won't see behavioral changes for HTTP URLs

## Related

- New database connection string rule: #1099
- Basic auth rule documentation: https://github.com/secretlint/secretlint/blob/master/packages/%40secretlint/secretlint-rule-basicauth/README.md

Would appreciate feedback on the preferred approach\!
